analysis_driver/dataset.py

```diff
@@ -344,6 +346,23 @@ def _find_pooling_step_for_artifact(art, expected_pooling_step_name=None, max_it
             )
         return art.input_artifact_list()
 
+    @property
+    def rapid_samples_by_lane(self):
+        if self._rapid_samples_by_lane is None:
+            self._rapid_samples_by_lane = {}
+
+            flowcell = set(self.lims_run.parent_processes()).pop().output_containers()[0]
+            for lane, artifact in flowcell.placements.items():
+                if len(artifact.reagent_labels) > 1:
+                    continue  # we don't want to run rapid processing on pools
+
+                assert len(artifact.samples) == 1
+                sample = artifact.samples[0]
+                if sample.udf.get('Rapid Analysis'):
+                    self._rapid_samples_by_lane[lane.split(':')[0]] = sample
+
+        return self._rapid_samples_by_lane
```

Comment:
We'd need to go through the sample sheet to know what samples to query. We'd also need to figure out how to insert fake Lims data in the integration test. Added a Todo for now, and returning a dict so that at least we're returning strings